test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/ProxyEndToEndTests.cs

```diff
@@ -0,0 +1,451 @@
+            var body = await response.Content.ReadAsStringAsync();
+            Assert.True(string.IsNullOrEmpty(body));
+            Assert.Equal(response.StatusCode, HttpStatusCode.NotModified);
+            Assert.Equal(response.Content.Headers.GetValues("Content-Type").ToArray()[0], "fake/custom");
+            Assert.True(response.Headers.Contains("Test"));
+        }
+
+        [Fact]
+        //"GET" request to proxy. backend returns 304 with no body so content-type should be null
+        public async Task EmptyGetDoesntReturnsContentType()
+        {
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, $"contentTypePresenceTest");
+            request.Headers.Add("return_empty_body", "1");
+            HttpResponseMessage response = await _fixture.HttpClient.SendAsync(request);
+            var body = await response.Content.ReadAsStringAsync();
+            Assert.True(string.IsNullOrEmpty(body));
+            Assert.Equal(response.StatusCode, HttpStatusCode.NotModified);
+            Assert.False(response.Content.Headers.Contains("Content-Type"));
+            Assert.True(response.Headers.Contains("Test"));
+        }
+
+
+        public class TestFixture : IDisposable
+        {
+            private readonly TestServer _testServer;
+            private readonly string _testHome;
+
+            public TestFixture()
+            {
+                // copy test files to temp directory, since accessing the metadata APIs will result
+                // in file creations (for test data files)
+                var scriptSource = Path.Combine(Environment.CurrentDirectory, @"..\..\..\TestScripts\Proxies");
+                _testHome = Path.Combine(Path.GetTempPath(), @"ProxyTests");
+                var scriptRoot = Path.Combine(_testHome, @"site\wwwroot");
+                FileUtility.CopyDirectory(scriptSource, scriptRoot);
+
+                HostOptions = new ScriptApplicationHostOptions
+                {
+                    IsSelfHost = true,
+                    ScriptPath = scriptRoot,
+                    LogPath = Path.Combine(_testHome, @"LogFiles\Application\Functions"),
+                    SecretsPath = Path.Combine(_testHome, @"data\Functions\Secrets"),
+                    TestDataPath = Path.Combine(_testHome, @"data\Functions\SampleData")
+                };
+
+                FileUtility.EnsureDirectoryExists(HostOptions.TestDataPath);
+
+                var optionsMonitor = TestHelpers.CreateOptionsMonitor(HostOptions);
+
+                var workerOptions = new LanguageWorkerOptions
+                {
+                    WorkerConfigs = TestHelpers.GetTestWorkerConfigs()
+                };
+
+                var provider = new HostFunctionMetadataProvider(optionsMonitor, NullLogger<HostFunctionMetadataProvider>.Instance, new TestMetricsLogger());
+
+                var builder = AspNetCore.WebHost.CreateDefaultBuilder()
```

Comment:
Pulling your branch, I see these tests don't currently run because the fixture is set up incorrectly. The tests have been disabled for a long time, so likely everything got stale. I've created a branch https://github.com/Azure/azure-functions-host/pull/8602 where I got most of these tests running. Please update your branch accordingly.